### DISCUSSION TITLE
A second rotary encoder to control directly the Algorithm

### DISCUSSION THREAD

**Discussion by Spacepadrille** [probonopd/MiniDexed#725] [Q&A] [answered]
Is there a way to wire a second rotary encoder with only one function : select the algorithm of the selected TG (no need of visual on the screen, just run thru the 32 algo) ?
The algorithm setting if far deep in the menu, it would make easier to sound design. 
By the way, what else can we control with midi buttons or rotary encoders ? I saw the general midi commands such general volume, cut off and resonance. Are there others ?

**Comment by probonopd** [comment, marked as answer]
Hello @Spacepadrille, welcome to MiniDexed.

The rotary encoder is meant for menu operation as described on
https://github.com/probonopd/MiniDexed/wiki/Menu

It is not planned to have more than one rotary encoder. But you can build your own little controller that contains a rotary encoder and sends a MIDI sysex message to change the algorithm when it is turned.

**Reply by Spacepadrille** [reply]
Thanks a lot for the answer ! I'm amazed by MiniDexed and spend a lot of time on it last days.
If I understand well what you said, there is a way to send MIDI sysex to a choosen parameter directly from a DIY controller. I suppose MIDI sysex is different than midi cc, I will make some research on it. Is there a table with some MIDI sysex addresses to reach parameters, e.g. algorithm ? Sorry I'm new in coding, assembling etc..., this is a huge domain.

**Reply by probonopd** [reply]
Yes, check the original DX7 documentation - we are using the same sysex.

**Reply by Spacepadrille** [reply]
Thanks a lot, will do this.
So in theory we can build a DIY MidiSisEx controller mimicking all the original faders, buttons etc of the original DX7 ?  ;-)

**Reply by probonopd** [reply]
Not only those. But eight times the Jellinghaus DX-Programmer.

![image](https://github.com/user-attachments/assets/8601009b-4171-4194-8244-197688b120b3)

**Reply by Spacepadrille** [reply]
A deep rabbit hole just popped in front of me.... Not sure I have to thank you :-)
